frontend/src/pages/spotnet/dashboard/withdraw-all/WithdrawAll.jsx

```diff
@@ -0,0 +1,70 @@
+import React, { useState } from 'react';
+import BalanceCards from 'components/BalanceCards';
+import { ReactComponent as ETH } from '../../../../assets/icons/ethereum.svg';
+import { ReactComponent as USDC } from '../../../../assets/icons/borrow_usdc.svg';
+import { ReactComponent as STRK } from '../../../../assets/icons/strk.svg';
+import './withdraw_all.css';
+import { useWalletStore } from 'stores/useWalletStore';
+import Button from 'components/ui/Button/Button';
+import { ReactComponent as AlertHexagon } from 'assets/icons/alert_hexagon.svg';
+import { axiosInstance } from 'utils/axios';
+
+const WithdrawAll = () => {
+  const { walletId } = useWalletStore();
+  const [balances, setBalances] = useState([
+    { icon: <ETH />, title: 'ETH', balance: '0.00' },
+    { icon: <USDC />, title: 'USDC', balance: '0.00' },
+    { icon: <STRK />, title: 'STRK', balance: '0.00' },
+  ]);
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState(null);
+
+  const handleWithdrawAll = async () => {
+    setLoading(true);
+    setError(null);
+
+    try {
+      const response = await axiosInstance.get(`/api/withdraw-all?wallet_id=${walletId}`);
+      console.log(response);
+      if (response.status === 200) {
+        console.log('Withdraw All operation completed successfully.');
+      } else {
+        console.log('Failed to complete the Withdraw All operation.');
+      }
+    } catch (error) {
+      console.error('Error during Withdraw All operation:', error);
+      setError('Something went wrong. Please try again.');
+    } finally {
+      setLoading(false);
+    }
+  };
```

Comment:
please use `useMutation` of `tanstack-query` we already use, extract this functionality to a custom hook called `useWithdrawAll`, you can return loading state (`isPending`) to a component, and show error using `notify` with "error" type in `onError` callback of `useMutation`, don't need to render error in component, just show it in a toast.

Also I believe it is enough to have try/catch, checking for 200 is redundand with axios, also clean up console logs.

Reply:
Updated!